client/my-sites/stats/stats-module/all-time-nav.jsx

```diff
@@ -0,0 +1,57 @@
+/**
+ * External dependencies
+ **/
+import React from 'react';
+import { connect } from 'react-redux';
+import { flowRight } from 'lodash';
+import { localize, moment } from 'i18n-calypso';
+import page from 'page';
+
+/**
+ * Internal dependencies
+ */
+import SelectDropdown from 'components/select-dropdown';
+import { recordGoogleEvent } from 'state/analytics/actions';
+import { getSelectedSiteId } from 'state/ui/selectors';
+import { getSiteSlug } from 'state/sites/selectors';
+
+export const StatsModuleSummaryLinks = props => {
+	const { translate, path, siteSlug, query } = props;
+
+	const now = moment();
+	const quarter = moment().startOf( 'quarter' );
+	const daysSinceQuarterStart = now.diff( quarter, 'd' );
+	const options = [
+		{ value: '0', label: translate( 'Summary' ) },
+		{ value: '7', label: translate( 'Last 7 days' ) },
+		{ value: '30', label: translate( 'Last 30 days' ) },
+		{ value: `${ daysSinceQuarterStart }`, label: translate( 'the past quarter' ) },
+		{ value: '-1', label: translate( 'All Time' ) }
+	];
+
+	const onSelect = ( item ) => {
+		if ( item.value !== '0' ) {
```

Comment:
Indeed this a bit of a hack - I'm using that value as a label essentially.  Perhaps I could show the current period name instead?  So if on 'Day' it would say 'Day Summary' and that link would redirect to the summary page.